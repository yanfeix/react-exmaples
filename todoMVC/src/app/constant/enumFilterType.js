const FilterType = {
    ALL: 'SHOW_ALL',
    ACTIVE: 'SHOW_ACTIVE',
    COMPLETED: 'SHOW_COMPLETED',
};

export default FilterType;
