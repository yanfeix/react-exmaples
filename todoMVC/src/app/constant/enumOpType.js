const OpType = {
    ADD: 'ADD',
    DELETE: 'DELETE',
    EDIT: 'EDIT',
    ACTIVATE: 'ACTIVATE',
    COMPLETE: 'COMPLETE',
    CLEAR_COMPLETED: 'CLEAR COMPLETED',
};

export default OpType;
