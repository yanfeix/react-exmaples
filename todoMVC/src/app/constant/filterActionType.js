const FilterAction = {
    SET_FILTER_TYPE: 'SET_FILTER_TYPE',
};

export default FilterAction;
