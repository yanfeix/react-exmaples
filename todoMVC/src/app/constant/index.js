export { default as OpType } from './enumOpType';
export { default as FilterType } from './enumFilterType';
export { default as UserAction } from './userActionType';
export { default as FilterAction } from './filterActionType';
