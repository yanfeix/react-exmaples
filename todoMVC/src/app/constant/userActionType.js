const UserAction = {
    ADD: 'ADD_TODO',
    REMOVE: 'REMOVE_TODO',
    MODIFY: 'MODIFY_TODO',
    ACTIVATE: 'ACTIVATE_TODO',
    COMPLETE: 'COMPLETE_TODO',
    CLEAR_COMPLETED: 'CLEAR_COMPLETED',
    TOGGLE: 'TOGGLE_ALL',
};

export default UserAction;
