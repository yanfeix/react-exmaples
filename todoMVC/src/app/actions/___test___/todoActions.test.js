import uuid from 'uuid';
import { UserAction } from '../../constant';
import {
    addTodo, removeTodo, modifyTodo, activateTodo,
    completeTodo, clearCompletedTodo, toggleAll,
} from '../todoActions';

describe(' Todo CRUD Actions ', () => {
    const todo = {
        id: uuid(),
        name: 'Working on UT',
        active: true,
        createdTime: Date.now(),
    };

    it(' add todo to return an action object { type, payload } ', () => {
        const text = 'Working on Unit Testing';
        expect(addTodo(text)).toMatchSnapshot({
            payload: {
                id: expect.any(String),
                createdTime: expect.any(Number), // expect.any(Date) does not work here
            },
        });
    });

    it(' remove todo to return an action object { type, payload }', () => {
        const obj = {
            type: UserAction.REMOVE,
            payload: { ...todo },
        };
        expect(removeTodo(todo)).toMatchObject(obj);
    });

    it(' modify Todo, return an action object { type, payload }', () => {
        const newVal = 'Filling timesheet on Friday';
        const obj = {
            type: UserAction.MODIFY,
            payload: { todo, newVal },
        };
        expect(modifyTodo(todo, newVal)).toMatchObject(obj);
    });

    it(' activate Todo, return an action object { type, payload }', () => {
        const obj = {
            type: UserAction.ACTIVATE,
            payload: { ...todo },
        };
        expect(activateTodo(todo)).toMatchObject(obj);
    });

    it(' complete todo, return an action object { type, payload }', () => {
        const obj = {
            type: UserAction.COMPLETE,
            payload: { ...todo },
        };
        expect(completeTodo(todo)).toMatchObject(obj);
    });

    it(' clear completed Todo, return an action object { type }', () => {
        const obj = {
            type: UserAction.CLEAR_COMPLETED,
        };
        expect(clearCompletedTodo()).toMatchObject(obj);
    });

    it(' toggle all todos, return an action object { type, payload }', () => {
        const objChecked = {
            type: UserAction.TOGGLE,
            payload: { checked: true },
        };
        const objUnchecked = {
            type: UserAction.TOGGLE,
            payload: { checked: false },
        };
        expect(toggleAll(true)).toMatchObject(objChecked);
        expect(toggleAll(false)).toMatchObject(objUnchecked);
    });
});
