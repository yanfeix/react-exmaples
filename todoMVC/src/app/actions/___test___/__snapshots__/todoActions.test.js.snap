// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Todo CRUD Actions   add todo to return an action object { type, payload }  1`] = `
Object {
  "payload": Object {
    "active": true,
    "createdTime": Any<Number>,
    "id": Any<String>,
    "name": "Working on Unit Testing",
  },
  "type": "ADD_TODO",
}
`;
